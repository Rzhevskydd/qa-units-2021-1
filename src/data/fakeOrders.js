const fakeOrders = [
	{
		id: 123,
		date: 1544356800000,
		shop: 'Alihandro Express',
		items: [
			'Утиный пластмасса для показ новый год',
			'Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан',
			'Новый стиль один розница яйцо для упаковки форма латекс',
		]
	},
	{
		id: 124,
		date: 1552481120000,
		shop: 'Lamodник.ru',
		items: [
			'Жакет - BOREAL5',
			'Miss Gabby Костюм',
			'Ostin перчатки мужские',
			'Zara худи роз.',
		]
	},
	{
		id: 126,
		date: 1552585550000,
		shop: 'Эльдоградо',
		items: [
			'Ноутбук Apple MacBook Air 13.3" (MQD32RU/A)',
			'Игровая приставка Sony PlayStation 4 Pro 1TB Black (CUH-7208B)',
		]
	},
];

const it1 = {items: ['item1', 'item2']}
const it2 = {items: ['1', '2', '3']}
let myFakeOrders = [it1, it2]

export {
	fakeOrders,
	it1,
	it2,
	myFakeOrders
}
