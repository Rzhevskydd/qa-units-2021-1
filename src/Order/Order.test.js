import React from 'react'

import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Order from './Order.js';
import {fakeOrders} from "../data/fakeOrders";

jest.mock('../utils/getDate');
import {getDate} from '../utils/getDate';

// getDate.mockReturnValue('31 февраля, чт, 2007 год');

configure({adapter: new Adapter()});

const it1 = {items: ['item1', 'item2']}
const it2 = {items: ['1', '2', '3']}
let orders = [it1, it2]

let fakeOrder = Object.assign({}, fakeOrders[0]);

describe('Order.js', () => {

  beforeEach(() => {
    getDate.mockReturnValue('31 февраля, чт, 2007 год');
    fakeOrder = Object.assign({}, fakeOrders[0]);
  });
  afterAll(() => {
    jest.resetModules();
  });

  it('render no order is null', () => {
    const wrapper = shallow(<Order />);
    expect(wrapper.getElement()).toBeNull();
  })

  it('render with null props in data', () => {
    const data = {
      order: {
        shop: null,
        date: null,
      },
    };

    const wrapper = shallow(<Order {...data}/>);

    expect(wrapper.getElement()).toBeNull();
  });

  it('render with items null in data', () => {
    fakeOrder.items = null

    const wrapper = shallow(<Order order={fakeOrder}/>);
    expect(getDate).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('render fakeOrder[0] snapshot', () => {
    const order = fakeOrders[0]
    const wrapper = shallow(<Order order={order}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});

