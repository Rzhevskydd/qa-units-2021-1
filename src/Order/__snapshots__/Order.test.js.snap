// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render fakeOrder[0] snapshot 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      31 февраля, чт, 2007 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order.js render with items null in data 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      31 февраля, чт, 2007 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;
